apps/zk-explorer/containers/ProjectDetailTabsContainer/Circuits/client/NavSideBar.tsx

```diff
@@ -1,7 +1,134 @@
+      >
+        <div className="p-2">
+          <SideBarLine className="dark:!fill-mono-0" />
+        </div>
+        {!isCollapsed && (
+          <Typography
+            variant="body2"
+            fw="bold"
+            className="!text-mono-200 dark:!text-mono-0"
+          >
+            Files
+          </Typography>
+        )}
+      </div>
+
+      {/* Files Tree */}
+      <div className={cx({ hidden: isCollapsed })}>
+        <UncontrolledTreeEnvironment<FileType>
+          dataProvider={new StaticTreeDataProvider(filesData)}
+          getItemTitle={(item) => item.data.fileName}
+          // disableMultiselect={true}
+          onFocusItem={(item) => {
+            if (!item.isFolder) {
+              handleFileSelect(item.index.toString());
+            }
+          }}
+          viewState={{
+            'file-tree': {
+              expandedItems: activeFileIndex
+                ? getExpandedFoldersFromActiveFileIndex(activeFileIndex)
+                : undefined,
+              selectedItems: activeFileIndex ? [activeFileIndex] : undefined,
+            },
+          }}
+          renderItemTitle={({ item, context }) => (
+            <div className="w-full flex items-center justify-between gap-2 overflow-hidden">
+              <div className="flex items-center gap-2">
+                {item.isFolder ? (
+                  context.isExpanded ? (
+                    <FolderOpenFill className="!fill-mono-120" />
+                  ) : (
+                    <FolderFill className="!fill-mono-120" />
+                  )
+                ) : (
+                  <FileLine className="!fill-mono-120" />
+                )}
+                <Typography variant="body2" className="text-ellipsis">
+                  {item.data.fileName}
+                </Typography>
+              </div>
+              <div>
+                {item.data.isTrustedSetup && (
+                  <ShieldedCheckLineIcon className="!fill-mono-120" />
+                )}
+              </div>
+            </div>
+          )}
+          renderItemArrow={({ item, context }) =>
+            item.isFolder ? (
+              <div
+                className="rct-tree-item-arrow-isFolder rct-tree-item-arrow"
+                {...context.arrowProps}
+                // {...context.interactiveElementProps.onClick}
+              >
+                <ChevronDown
+                  className={cx({
+                    '-rotate-90': !context.isExpanded,
+                  })}
+                  enableBackground="new 0 0 16 16"
+                  xmlSpace="preserve"
+                />
+              </div>
+            ) : null
+          }
+        >
+          <Tree treeId="file-tree" rootItem="root" />
+        </UncontrolledTreeEnvironment>
+      </div>
+    </div>
+  );
 };
 
 export default NavSideBar;
+
+/** @internal */
+function getExpandedFoldersFromActiveFileIndex(activeFileIndex: string) {
```

Comment:
Consider wrapping the `getExpandedFoldersFromActiveFileIndex` within `useMemo`. This optimization is recommended because the function's output only depends on the `activeFileIndex`. Since the same `activeFileIndex` (e.g., someFile.txt) might be selected multiple times, using `useMemo` ensures that the calculation to return the expanded folders is only performed when the `activeFileIndex` actually changes.

```
const expandedFolders = useMemo(() => 
    getExpandedFoldersFromActiveFileIndex(activeFileIndex), 
    [activeFileIndex]
);
```